src/redux/books/books.js

```diff
@@ -32,9 +29,55 @@ const booksReducer = (state = stateInit, action) => {
     case REMOVE_BOOK:
       return state.filter((book) => book.id !== action.payload);
 
+    case GET_BOOKS:
+      return action.payload;
+
     default:
       return state;
   }
 };
 
+export const getBooksList = () => async (dispatch) => {
+  const booksList = await fetch(`${url}`)
+    .then((response) => response.json());
+  const booksID = Object.keys(booksList);
+  const booksArr = [];
+  booksID.map((item) => booksArr.push({
+    id: item,
+    title: booksList[item][0].title,
+    author: booksList[item][0].author,
+    category: booksList[item][0].category,
+  }));
+  dispatch(getBooks(booksArr));
+};
+
+export const postBook = (newBook) => async (dispatch) => {
+  await fetch(`${url}`, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: newBook.id,
+      title: newBook.title,
+      author: newBook.author,
+      category: newBook.category,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(addBook(newBook));
+};
```

Comment:
If I add a new book, I can clearly see that is straight away added to the list. But unfortunately, it is not persisting; on refresh, it disappears.

- [ ] Kindly check what exactly the issue is. First, you can check if it is added to the API. If it is, let's see why you are not getting it after refresh.